### PR TITLE
Implemented PBC and ECP.

### PR DESCRIPTION
**Pull request by kousuke-nakano**
Implemented the following functions in `pyscf/tools/trexio.py`. See also #153.

[Writing]
- `_scf_to_trexio`: Complex wavefunctions and twisted-average are done.

[Reading]
- `mol_from_trexio`: The ECP implementation `if trexio.has_ecp(tf)` is done.
- `scf_from_trexio`: UHF and ROHF are also supported.

[Test suites]
- Implemented many new tests for this `trexio` tool.

### CODE REVIEW

**Comment by kousuke-nakano** [comment]
I have switched it to Draft because it was ok with `python=3.8` and `python=3.12`, but not with `python=3.10`. I am investigating this.

**Comment by kousuke-nakano** [comment]
Dear @sunqm, I could not reproduce the failure of the check (Only `linux-build (3.10)` failed) at my local environment with the same Python version `3.10.18`. I would appreciate it if you could relaunch the workflow, or am I allowed to trigger it again?

File: pyscf/tools/trexio.py [hunk 1]
```diff
@@ -210,52 +213,180 @@ def _mol_to_trexio(mol, trexio_file):
...
     _mol_to_trexio(mol, trexio_file)
 
-    # 2.3 Periodic boundary calculations (pbc group)
+    # PBC
     if isinstance(mol, pbc.gto.Cell):
-        kpt = mf.kpt
-        kpt = mol.get_scaled_kpts(kpt)
-        if np.all(np.array(kpt) == 0.0):
-            # gamma point PBC. Real wavefunction
+        kpts = np.array(mf.kpts)
```
[the remark this answers is not in the document]
Comment (kousuke-nakano) [reply pyscf/tools/trexio.py]: @sunqm, Thank you very much for the review. Yes, this is a corner case that I have overlooked. I have fixed it and introduced corresponding new tests `test_mf_k_single_grid_rhf_ae_6_31g` and `test_mf_k_single_grid_uhf_ae_6_31g`. They failed with the previous version of `trexio.py`, while they pass right now.

**Review by sunqm — APPROVED** [review]
